lib/logging/log.js

```diff
@@ -0,0 +1,415 @@
+/*!
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * @module logging/log
+ */
+
+'use strict';
+
+var arrify = require('arrify');
+var extend = require('extend');
+var is = require('is');
+var JSONStream = require('JSONStream');
+var nodeutil = require('util');
+var pumpify = require('pumpify');
+var streamEvents = require('stream-events');
+var through = require('through2');
+
+/**
+ * @type {module:common/serviceObject}
+ * @private
+ */
+var ServiceObject = require('../common/service-object.js');
+
+/**
+ * A log is a named collection of records or entries, each entry representing a
+ * timestamped event. Logs can be produced by Google Cloud Platform services, by
+ * third-party services, or by your applications. For example, the log
+ * `apache-access` is produced by the Apache Web Server, but the log
+ * `compute.googleapis.com/activity_log` is produced by Google Compute Engine.
+ *
+ * @resource [Introduction to Logs]{@link https://cloud.google.com/logging/docs/api/#logs}
+ *
+ * @alias module:logging/log
+ * @constructor
+ *
+ * @param {object} options - [Configuration object](#/docs/?method=gcloud).
+ *
+ * @example
+ * var gcloud = require('gcloud')({
+ *   keyFilename: '/path/to/keyfile.json',
+ *   projectId: 'grape-spaceship-123'
+ * });
+ *
+ * var logging = gcloud.logging();
+ * var log = logging.log('my-log');
+ */
+function Log(logging, name) {
+  var methods = {
+    /**
+     * Delete the log.
+     *
+     * @resource [projects.logs.delete API Documentation]{@link https://cloud.google.com/logging/docs/api/ref/rest/v1beta3/projects.logs/delete}
+     *
+     * @param {function=} callback - The callback function.
+     * @param {?error} callback.err - An error returned while making this
+     *     request.
+     * @param {object} callback.apiResponse - The full API response.
+     *
+     * @example
+     * log.delete(function(err, apiResponse) {
+     *   if (!err) {
+     *     // The log was deleted.
+     *   }
+     * });
+     */
+    delete: true
+  };
+
+  this.name = name.replace(/^.*logs\//, '');
+  this.unformattedName = name;
+
+  ServiceObject.call(this, {
+    parent: logging,
+    baseUrl: '/logs',
+    id: this.name,
```

Comment:
This relates to how requests are patched upwards through our library, continuously appending pieces of the URL. I'm only passing `name` here, as it's already being appended to:

`...com/v1beta3/projects/{projectId}/logs/`

A user can refer to their log only by name ('my-log'), instead of a fully qualified name including the project ID, and it will work the same.

RE: `unformattedName` good catch, will remove.